### DISCUSSION TITLE
Retriggeable RuleTimer

### DISCUSSION THREAD

**Discussion by jnherm** [arendst/Tasmota#14238] [Q&A] [answered]
Does anyone have implemented re-triggerable ruletimer on their rules? Please help if you have.

**Comment by sfromis** [comment, marked as answer]
From the rule triggering (`Rules#Timer`) on the expiry on the RuleTimer, you can can set it up again using the RuleTimer command, including the interval. Upon expiry, Tasmota does not remember anything about the old timer.

**Reply by barbudor** [reply]
If by retriggering you mean Restarting a count down before the expiracy, this also works fine.
Whether the timer is running or ended, the RuleTimer command start a new cycle

**Reply by jnherm** [reply]
What I needed is if before the expiry of 4sec, whenever there is another button trigger, I will add again another 4sec. So if currently it has still 2sec left, after another button trigger then it will become 6sec.

**Reply by sfromis** [reply]
Anytime you issue the RuleTimer command, the interval becomes the starting point for the countdown, overriding any existing countdown.

Note that the PulseTime command is simpler to use if you want to keep a relay/light or similar on for for an interval, including extending the interval in case of a new signal like a button, anything powering the relay on again will restart the interval.
